src/table/cargo_const.h

```diff
@@ -7,157 +7,75 @@
 
 /** @file cargo_const.h Table of all default cargo types */
 
-/** Construction macro for a #CargoSpec structure. */
-#define MK(bt, label, c, e, f, g, h, fr, te, ks1, ks2, ks3, ks4, ks5, l, m, cmult) \
-		{bt, label, c, c, e, cmult, f, {g, h}, fr, te, 0, 0, ks1, ks2, ks3, ks4, ks5, l, m, nullptr, nullptr, 0}
-/** Cargo types available by default. */
-static const CargoSpec _default_cargo[] = {
-	MK(  0, 'PASS', 152,  1, 3185,  0,  24, false, TE_PASSENGERS,
-		STR_CARGO_PLURAL_PASSENGERS,     STR_CARGO_SINGULAR_PASSENGER,      STR_PASSENGERS, STR_QUANTITY_PASSENGERS,   STR_ABBREV_PASSENGERS,
-		SPR_CARGO_PASSENGER,     CC_PASSENGERS,                    0x400),
-
-	MK(  1, 'COAL',   6, 16, 5916,  7, 255, true,  TE_NONE,
-		STR_CARGO_PLURAL_COAL,           STR_CARGO_SINGULAR_COAL,           STR_TONS,       STR_QUANTITY_COAL,         STR_ABBREV_COAL,
-		SPR_CARGO_COAL,          CC_BULK,                          0x100),
+/** Construction macros for the #CargoSpec StringID entries. */
+#define MK_STR_CARGO_PLURAL(label_plural) STR_CARGO_PLURAL_ ## label_plural
+#define MK_STR_CARGO_SINGULAR(label_singular) STR_CARGO_SINGULAR_ ## label_singular
+#define MK_STR_QUANTITY(label_plural) STR_QUANTITY_ ## label_plural
+#define MK_STR_ABBREV(label_plural) STR_ABBREV_ ## label_plural
+/** Construction macros for the #CargoSpec SpriteID entry. */
+#define MK_SPRITE(label_plural) SPR_CARGO_ ## label_plural
 
-	MK(  2, 'MAIL',  15,  4, 4550, 20,  90, false, TE_MAIL,
-		STR_CARGO_PLURAL_MAIL,           STR_CARGO_SINGULAR_MAIL,           STR_BAGS,       STR_QUANTITY_MAIL,         STR_ABBREV_MAIL,
-		SPR_CARGO_MAIL,          CC_MAIL,                          0x200),
+/*
+ * Construction macro for a #CargoSpec structure.
+ * The order of arguments matches the order in which they are defined in #CargoSpec.
+ * Some macros are used to automatically expand to the correct StringID consts, this
+ * means that adding/changing a cargo spec requires updating of the following strings:
+ * - STR_CARGO_PLURAL_<str_plural>
+ * - STR_CARGO_SINGULAR_<str_singular>
+ * - STR_QUANTITY_<str_plural>
+ * - STR_ABBREV_<str_plural>
+ * And the following sprite:
+ * - SPR_CARGO_<str_plural>
+ */
+#define MK(bt, label, colour, weight, mult, ip, td1, td2, freight, te, str_plural, str_singular, str_volume, classes) \
```

Comment:
I'd definitely consider documenting this "properly" with full doxygen treatment (`@param`)

Reply:
Added docstrings as far as #CargoSpec  was documented. I think documenting all of CargoSpec is outside the scope of this PR.